### ISSUE TITLE
How to load the checkpoint only the encoder part?

### ISSUE THREAD

**Issue by OswaldoBornemann** [k2-fsa/icefall#1399]
How to load the checkpoint only the encoder part?

**Comment by OswaldoBornemann** [comment]
Thank you.

**Comment by OswaldoBornemann** [comment]
Also, can i only load the parameters from the encoder in the checkpoint without loading the optimizer, scheduler, grad_scaler, or sampler? Will this hurt the performance?

**Comment by marcoyang1998** [comment]
You can have a look at the code below, which loads weights from a checkpoint by matching the prefix. It's used for fine-tuning a pre-trained model on new data, and normally we won't use the optimizer/scheduler/grad_scaler state in the old checkpoint.

https://github.com/k2-fsa/icefall/blob/735fb9a73dea7d27e95056add6598ae7a282d6f9/egs/librispeech/ASR/pruned_transducer_stateless7/finetune.py#L626

**Comment by OswaldoBornemann** [comment]
> You can have a look at the code below, which loads weights from a checkpoint by matching the prefix. It's used for fine-tuning a pre-trained model on new data, and normally we won't use the optimizer/scheduler/grad_scaler state in the old checkpoint.
> 
> https://github.com/k2-fsa/icefall/blob/735fb9a73dea7d27e95056add6598ae7a282d6f9/egs/librispeech/ASR/pruned_transducer_stateless7/finetune.py#L626

Thank you very much.

**Comment by OswaldoBornemann** [comment]
@csukuangfj @marcoyang1998 I would like to ask one more thing. When we want to fine-tune the model on the new dataset, is it enough for us only to load the parameter of the encoder from the pre-trained model? Or do me need to load the parameter of the joiner model as well?
